runtime/client/ext-helper.js

```diff
@@ -36,22 +35,24 @@ function main (target, runner) {
   keepAlive()
   getActivityDescriptor()
     .then(descriptor => {
+      aliveInterval.unref()
       translator.setLogger(require('logger')(`@ipc-${process.pid}`))
-      var activity = translator.translate(descriptor)
-      activity.appId = appId
-      activity.appHome = target
-      global[apiSymbol] = activity
+      var api = translator.translate(descriptor)
+      api.appId = appId
+      api.appHome = target
+      global[apiSymbol] = api
 
       /**
        * Executes app's main function
        */
-      launchApp(main, activity)
-      runner(appId, pkg, activity)
+      launchApp(main, api)
 
       process.send({
         type: 'status-report',
         status: 'ready'
       })
+
+      runner(appId, pkg)
```

Comment:
Why do you move the runner after sending `ready` report?

Reply:
Ready event should be sent to VUI to reset ANR as soon as possible. And after that normal ANR would work instead.